ScrapeRepos.cjs

```diff
@@ -0,0 +1,243 @@
+const axios = require('axios');
+const cheerio = require('cheerio');
+const { time } = require('console');
+const showdown = require('showdown');
+const readline = require('readline').createInterface({
+  input: process.stdin,
+  output: process.stdout
+});
+
+
+async function getUserInput() {
+  return new Promise(resolve => {
+    readline.question('From which timeframe do you want to get your data (0: daily; 1: weekly; 2: monthly): ', input => {
+      resolve(input);
+    });
+  });
+}
+
+/** Fetches the repo owner and name for each repo on the GitHub trending page
+ * @param userInput 0: daily; 1: weekly; 2: monthly
+ * @returns an array that stores alternatingly the owner and the name of each repo
+ */
+async function fetchRepos(userInput) {
+  try {
+    const timeModes = ['?since=daily', '?since=weekly', '?since=monthly'];
+    const response = await axios.get('https://github.com/trending' + timeModes[userInput]);
+    const $ = cheerio.load(response.data);
+    const repos = [];
+
+    $('h2 a').each((i, el) => {
+      const repoName = $(el).text().trim();
+      repos.push(repoName);
+    });
+    
+    var trendingSplit = [];
+    // trim the repos to be correctly formatted
+    repos.forEach((repo) => {
+      let trimmedName = repo.replace(/\n\s+/g, '');
+      trimmedName = trimmedName.replace(/\//g, "");
+      const stringSplit = trimmedName.split(" ");
+      trendingSplit.push(stringSplit[0]);
+      trendingSplit.push(stringSplit[1]);
+    });
+
+    return trendingSplit;
+  } catch (error) {
+    console.error(error);
+  }
+}
+
+/**  This function imports the ReadMe.md file for a repository (if it can be located)
+* @param owner - owner of the repo
+* @param name - name of the repo
+* @returns a string containing the text of the repo
+*           returns null if the repo can't be located
+*/
+async function getReadme(owner, name) {
+  const readmePaths = [`https://raw.githubusercontent.com/${owner}/${name}/release/readme.md`,
+                         `https://raw.githubusercontent.com/${owner}/${name}/dev/README.rst`,
+                        `https://raw.githubusercontent.com/${owner}/${name}/main/README.md`,
+                        `https://raw.githubusercontent.com/${owner}/${name}/master/README.md`];
+ 
+  for(i = 0; i < readmePaths.length; i++){
+    try{
+      const response = await axios.get(readmePaths[i])
```

Comment:
This operation could be a function

Reply:
But do you think this would be helpful? I don't believe that this specific string transformation would be used by any other service or class

Reply:
This could help another developer who will develop something in this class. Sometimes functions are used to make code more readable. My purpose at this review was about readability.

Reply:
I will also implement this in next PR